pyobis/dataset/dataset.py

```diff
@@ -95,6 +106,35 @@ def search(
         data = dataset.search(taxonid=res['worms_id']).execute()
     """  # noqa: E501
     url = obis_baseurl + "dataset"
+
+    # Keyword-only metadata search path (Elasticsearch simple_query_string via `q`)
+    if keyword is not None:
+        # Disallow combining keyword with other filters (limit/offset still allowed)
+        if any(
+            v is not None
+            for v in [
+                scientificname,
+                taxonid,
+                nodeid,
+                startdate,
+                enddate,
+                startdepth,
+                enddepth,
+                geometry,
+                flags,
```

Comment:
not a blocker but what if we do a negative membership check? since the allow list is considerably larger than exclusion list. if we add more request params to the method the list will keep on growing and an added effort on adding the param here too.

Reply:
The implementation is a little hairy so I put it in a function. I agree this is easier to maintain going forward. :+1: 

https://github.com/7yl4r/pyobis/commit/7c99d2f2d7e10fc017c6841f1102688cc57d4e01